pkg/relayer/proxy/server_builder.go

```diff
@@ -26,10 +28,30 @@ func (rp *relayerProxy) BuildProvidedServices(ctx context.Context) error {
 		return err
 	}
 
-	// Get the supplier's advertised information from the blockchain
-	supplier, err := rp.supplierQuerier.GetSupplier(ctx, supplierAddress.String())
-	if err != nil {
-		return err
+	var supplier sharedtypes.Supplier
+
+	// Wait for the supplier to be staked before building the relay servers.
+	for {
+		// Get the supplier's advertised information from the blockchain
+		supplier, err = rp.supplierQuerier.GetSupplier(ctx, supplierAddress.String())
+
+		// If the supplier is not found, wait for the supplier to be staked.
+		if err != nil && suppliertypes.ErrSupplierNotFound.Is(err) {
+			rp.logger.Info().Msgf(
+				"Waiting for the supplier with address %s to stake",
```

Comment:
Is it worth mentioning that the next check will be in 5 seconds? At which point maybe that value should get its own variable.